### PR TITLE
IBX-4962: React App page block added

### PR DESCRIPTION
**Pull request by julitafalcondusza**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | (https://issues.ibexa.co/browse/IBX-4962)
| Versions      |master
| Edition       | Commerce

React App page block added in Pages.

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date
- [x] Code samples are working
- [x] PHP code samples have been fixed with PHP CS fixer
- [x] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/content_management/pages/react_app_block.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
+---
+description: Create a block that allows an editor to embed a preconfigured React application into a page.
```
Comment (Gengar-i) [inline docs/content_management/pages/react_app_block.md:2]: I am not sure if we want to use React application or just React component, but maybe it is just synonym

File: docs/content_management/pages/react_app_block.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+Then, create a .js file in `/page-builder/react/blocks` directory.
+This file imports and exports the component: 
+
+``` js
+import Calculator from "../../../vendor/ibexa/page-builder/components/Calculator";
+
+export default {
+    Calculator: Calculator,
+
```
Comment (Gengar-i) [inline docs/content_management/pages/react_app_block.md:112]:
```suggestion
```

**Review by Gengar-i — APPROVED** [review]

File: docs/content_management/pages/react_app_block.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+| `visible` | (Optional) Toggles the block's visibility in the Page Builder elements menu. Remove the block from the layout before you publish another version of the page. |
+| `attributes` | (Optional) List of [block attributes](page_block_attributes.md). |
+
+For example:
+
+``` yaml
+ibexa_fieldtype_page:
+    react_blocks:
+        calculator:
+          name: Calculator (React)
```
Comment (Gengar-i) [inline docs/content_management/pages/react_app_block.md:42]: We have to describe for user that this name should be the same as Component name in /page-builder/react/blocks

File: docs/content_management/pages/react_app_block.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+## Create React App block
+
+In the following example you will learn how to create `Calculator` React App block.
+
+### Configure React App Block
+
+First, create a .jsx file which describes your component.
+You can create the file in any location.
+
+In the following example, create `Calculator.jsx` file in `../../../vendor/ibexa/page-builder/components/` directory:
```
Comment (dew326) [inline docs/content_management/pages/react_app_block.md:91]: I'm not sure if this is a good idea to write in our documentation to modify files in vendor

File: docs/content_management/pages/react_app_block.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+```
+attributes:
+  a:
+    name: Attribute A
+    type: string
+    options:
+      ...
+```
+
+Apps that are used as blocks are defined by default in `react/controllers` direction.
```
Comment (dew326) [inline docs/content_management/pages/react_app_block.md:74]: it is?

File: docs/content_management/pages/react_app_block.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+The `assets/apps.js` file is pre-configured with this path.
```
Comment (dew326) [inline docs/content_management/pages/react_app_block.md:75]: I think it is separate file

File: docs/content_management/pages/react_app_block.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    options:
+      ...
+```
+
+The `assets/apps.js` file is pre-configured with this path.
+
+Apps that are registered this way must be configured and referenced in the 
+semantic configuration to be registered as blocks.
+
+Apps that are used as blocks are defined by default in `react/controllers` direction.
```
Comment (dew326) [inline docs/content_management/pages/react_app_block.md:79]: no? Don't know what this folder is?

File: docs/content_management/pages/react_app_block.md [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+```
+attributes:
+  a:
+    name: Attribute A
+    type: string
+    options:
+      ...
+```
+
+The `assets/apps.js` file is pre-configured with this path.
```
Comment (dew326) [inline docs/content_management/pages/react_app_block.md:74]: we don't have apps.js file?

File: docs/content_management/pages/react_app_block.md [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+| `name` | Name of the block used in the Page Builder interface. It has to be the same as Component name used in `/page-builder/react/blocks` directory.|
+| `category` | Category in the Page Builder elements menu that the block is shown in. |
+| `thumbnail` | Thumbnail used in the Page Builder elements menu. |
+| `visible` | (Optional) Toggles the block's visibility in the Page Builder elements menu. Remove the block from the layout before you publish another version of the page. |
+| `attributes` | (Optional) List of [block attributes](page_block_attributes.md). |
+
+For example:
+
+``` yaml
+ibexa_fieldtype_page:
```
Comment (dew326) [inline docs/content_management/pages/react_app_block.md:39]: Shouldn't we mention about adding block config from JSON file? @kisztof

**Review by dew326 — CHANGES_REQUESTED** [review]

**Review by dew326 — APPROVED** [review]

File: docs/content_management/pages/react_app_block.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+        calculator:
+          name: Calculator
+          category: Demo
+          thumbnail: /bundles/ibexaicons/img/all-icons.svg#date
+          component: Calculator
+          attributes:
+            a:
+              type: integer
+            b: integer
+    }
```
Comment (MagdalenaZuba) [inline docs/content_management/pages/react_app_block.md:50]: Where is the opening one: '{'?
Can you double-check code indentation, it looks uneven.

File: docs/content_management/pages/react_app_block.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+Parameters passed as props must be converted so that they can be used as the configured type in the app.
+
+## Create React App block
+
+In the following example you will learn how to create `Calculator` React App block.
+
+### Configure React App Block
+
+First, create a .jsx file which describes your component.
+You can create the file at any location.
```
Comment (MagdalenaZuba) [inline docs/content_management/pages/react_app_block.md:86]:
```suggestion
You can place it in any location.
```

**Review by MagdalenaZuba — CHANGES_REQUESTED** [review]

File: docs/content_management/pages/react_app_block.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+which adds component and props variables.
+
+It is general for all the blocks.
+
+[[% include 'snippets/page_block_cache_clear.md' %]]
+
+## React App Block configuration
+
+React App blocks are regular [Page blocks](page_blocks.md) and can be confirgured on field definition level as any other block.
+File has exactly the same structure as regular YAML [block configuration](create_custom_page_block/#configure-block), except:
```
Comment (MagdalenaZuba) [inline docs/content_management/pages/react_app_block.md:21]: broken link

File: docs/content_management/pages/react_app_block.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+Both name and attributes support a short syntax and a long one for specifics.
+
+`Attributes` defined without sub-keys use the key as the identifier and name, and the value as the type:
+
+```
+attributes:
+  b: integer
+```
+
+Sub-keys can be used to specify any of the usual [attributes configuration](page_block_attributes/#page-block-attributes) key:
```
Comment (MagdalenaZuba) [inline docs/content_management/pages/react_app_block.md:63]: broken link

**Review by MagdalenaZuba — APPROVED** [review]

**Pull request merged**